lib/riemann/tools/health.rb

```diff
@@ -240,47 +270,50 @@ def sunos_memory
       end
 
       def darwin_top
+        return @cached_data[:darwin_top] if @cached_data[:darwin_top]
+
         raw = `top -l 1 | grep -i "^\\(cpu\\|physmem\\|load\\)"`.chomp
-        @topdata = { stamp: Time.now.to_i }
+        topdata = {}
         raw.each_line do |ln|
           if ln.match(/Load Avg: [0-9.]+, [0-9.]+, ([0-9.])+/i)
-            @topdata[:load] = Regexp.last_match(1).to_f
+            topdata[:load] = Regexp.last_match(1).to_f
           elsif ln.match(/CPU usage: [0-9.]+% user, [0-9.]+% sys, ([0-9.]+)% idle/i)
-            @topdata[:cpu] = 1 - (Regexp.last_match(1).to_f / 100)
```

Comment:
Are these unrelated stylistic changes?

Reply:
I split this in 3 commits. This is part of the first which is expected to not change the behavior of the program but makes data-caching more abstract: instead of just having cache for `top(1)` on MacOS, we have a "global" cache that just cache the output of top on MacOS.

The other commits then use this to also cache the output of `uptime` and the parser result.

Reply:
Cool. 👍🏻